dapr/clients/grpc/_request.py

```diff
@@ -5,31 +5,89 @@
 Licensed under the MIT License.
 """
 
-from typing import Optional, Union, Dict
+from typing import Dict, Optional, Union
 
-from dapr.clients.grpc._helpers import MetadataTuple
 from google.protobuf.any_pb2 import Any as GrpcAny
 from google.protobuf.message import Message as GrpcMessage
 
 from dapr.clients.base import DEFAULT_JSON_CONTENT_TYPE
+from dapr.clients.grpc._helpers import (
+    MetadataDict,
+    MetadataTuple,
+    tuple_to_dict,
+    to_bytes,
+    to_str,
+    unpack
+)
 
 
-class InvokeServiceRequestData:
+class DaprRequest:
+    """A base class for Dapr Request.
+
+    This is the base class for Dapr Request. User can gets the metadata.
+
+    Attributes:
+        metadata(dict): A dict to include the headers from Dapr Request.
+    """
+    def __init__(self, metadata: MetadataTuple = ()):
+        self.metadata = metadata  # type: ignore
+
+    @property
+    def metadata(self) -> MetadataDict:
+        """Get metadata from :class:`DaprRequest`."""
+        return self.get_metadata(as_dict=True)  # type: ignore
+
+    @metadata.setter
+    def metadata(self, val) -> None:
+        """Sets metadata."""
+        if not isinstance(val, tuple):
+            raise ValueError('val is not tuple')
+        self._metadata = val
+
+    def get_metadata(self, as_dict: bool = False) -> Union[MetadataDict, MetadataTuple]:
+        """Gets metadata from the request.
+
+        Args:
+            as_dict (bool): dict type metadata if as_dict is True. Otherwise, return
+                tuple metadata.
+
+        Returns:
+            dict or tuple: request metadata.
+        """
+        if as_dict:
+            return tuple_to_dict(self._metadata)
+        return self._metadata
+
+
+class InvokeServiceRequest(DaprRequest):
```

Comment:
Change it to `ServiceRequest` to be consistent with `BindingRequest`?

Reply:
BindingRequest can be used for input and ouput binding so that I remove the prefix. InvokeServiceRequest is more intuitive than ServiceRequest. to me... we can change it later... this is intiail commit.